src/NServiceBus.Metrics.PerformanceCounters.Tests/PerformanceCounterUpdaterTests.cs

```diff
@@ -196,6 +198,45 @@ public void Timers_within_payload_should_be_converted_into_performance_counters(
       }
     ]
   }";
+        [Test]
+        public void Timers_within_payload_should_be_reported_as_zeros_after_specific_period_from_last_observed_pipeline_completion()
+        {
+            var resetTimersAfter = TimeSpan.FromMilliseconds(100);
+
+            var cache = new MockPerformanceCountersCache();
+            var sut = new PerformanceCounterUpdater(cache, new Dictionary<string, CounterInstanceName?>(), resetTimersAfter);
+
+            Thread.Sleep(TimeSpan.FromTicks(resetTimersAfter.Ticks * 2));
```

Comment:
Love tests with stuff like this in it! ;-)

Reply:
I'm not fan of this as well, but for this test only I don't want to introduce https://github.com/Fody/Ionad cannon.

Reply:
I know, totally agree with the test. This is so simple, no need to make it more complex.